webhook/workspace/handler/warning.go

```diff
@@ -0,0 +1,162 @@
+//
+// Copyright (c) 2019-2022 Red Hat, Inc.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package handler
+
+import (
+	"fmt"
+	"strings"
+
+	dwv2 "github.com/devfile/api/v2/pkg/apis/workspaces/v1alpha2"
+)
+
+type unsupportedWarnings struct {
+	serviceAnnotations  []string
+	endpointAnnotations []string
+	dedicatedPod        []string
+	imageComponent      []string
+	customComponent     []string
+	volumeSize          []string
+	eventPostStop       []string
+	eventPreStop        []string
+}
```

Comment:
Since these are basically sets (should have no duplicate component names in any field), you could consider using `map[string]bool` instead of `[]string` here. This would simplify their usage:

* Adding an element: `serviceAnnotations[component.Name] = true`
* Checking membership: `if serviceAnnotations[component.Name] {/*there is a warning for component.Name*/}
* Doing set subtraction (e.g. in `checkForAddedUnsupportedFeatures` below):
    ```go
    func getNewElements(old, new map) []string{
      var newNames []string
      for name := range new {
        if !old[name] {
          newNames = append(newNames, name)
        }
      }
      return newNames
    }
    ```

Reply:
Submitted a fixup to use a `map[string]bool` :+1: